### PR TITLE
Do not generate new address when a new payment is detected

### PR DESCRIPTION
**Pull request by NicolasDorier**
When a bitcoin address is used for a payment, it was being switched with a new address.

We got feedback of customers and merchants very confused about it, and thinking they got scammed.
I believe we should just reuse addresses. This isn't good for privacy, but this is a corner case that almost never happen. It isn't really worth bothering support with this.

Fix https://github.com/btcpayserver/btcpayserver/issues/4980

### CODE REVIEW

File: BTCPayServer.Tests/UnitTest1.cs [hunk 1]
```diff
@@ -2154,7 +2154,6 @@ public async Task InvoiceFlowThroughDifferentStatesCorrectly()
                 txFee = localInvoice.BtcDue - invoice.BtcDue;
                 Assert.Equal("paidPartial", localInvoice.ExceptionStatus.ToString());
                 Assert.Equal(1, localInvoice.CryptoInfo[0].TxCount);
```
Comment (Kukks) [inline BTCPayServer.Tests/UnitTest1.cs:2156]:
```suggestion
                Assert.Equal(1, localInvoice.CryptoInfo[0].TxCount);
                Assert.Equal(localInvoice.BitcoinAddress, invoice.BitcoinAddress); //Same address
```

**Review by Kukks — APPROVED** [review]
